modules/reactor/src/meshgenerators/PeripheralTriangleMeshGenerator.C

```diff
@@ -78,74 +84,54 @@ PeripheralTriangleMeshGenerator::validParams()
-      params.set<Real>("auto_area_func_default_size_dist") =
-          getParam<Real>("auto_area_func_default_size_dist");
-    if (isParamSetByUser("auto_area_function_num_points"))
-      params.set<unsigned int>("auto_area_function_num_points") =
-          getParam<unsigned int>("auto_area_function_num_points");
-    if (isParamSetByUser("auto_area_function_power"))
-      params.set<Real>("auto_area_function_power") = getParam<Real>("auto_area_function_power");
-    params.set<bool>("refine_boundary") = false;
-    params.set<std::vector<bool>>("refine_holes") = std::vector<bool>{false};
-    params.set<std::vector<bool>>("stitch_holes") = std::vector<bool>{true};
-    params.set<BoundaryName>("output_boundary") = _external_boundary_name;
-    params.set<SubdomainName>("output_subdomain_name") = _peripheral_ring_block_name;
-    params.set<MooseEnum>("tri_element_type") = getParam<MooseEnum>("tri_element_type");
-    addMeshSubgenerator("XYDelaunayGenerator", _input_name + "_periphery", params);
-    _build_mesh = &getMeshByName(_input_name + "_periphery");
+    pos.push_back(libMesh::Point());
+    ids.push_back(i);
   }
+  params.set<Real>("desired_area") = _desired_area;
+  params.set<std::string>("desired_area_func") = _desired_area_func;
+  params.set<bool>("use_auto_area_func") = getParam<bool>("use_auto_area_func");
+  if (isParamSetByUser("auto_area_func_default_size"))
+    params.set<Real>("auto_area_func_default_size") = getParam<Real>("auto_area_func_default_size");
+  if (isParamSetByUser("auto_area_func_default_size_dist"))
+    params.set<Real>("auto_area_func_default_size_dist") =
+        getParam<Real>("auto_area_func_default_size_dist");
+  if (isParamSetByUser("auto_area_function_num_points"))
+    params.set<unsigned int>("auto_area_function_num_points") =
+        getParam<unsigned int>("auto_area_function_num_points");
+  if (isParamSetByUser("auto_area_function_power"))
+    params.set<Real>("auto_area_function_power") = getParam<Real>("auto_area_function_power");
+  if (isParamValid("peripheral_ring_block_id"))
+    params.set<subdomain_id_type>("background_subdomain_id") =
+        getParam<subdomain_id_type>("peripheral_ring_block_id");
+  if (isParamValid("peripheral_ring_block_name"))
+    params.set<SubdomainName>("background_subdomain_name") =
+        getParam<SubdomainName>("peripheral_ring_block_name");
+  if (isParamValid("external_boundary_id"))
+    params.set<boundary_id_type>("external_boundary_id") =
+        getParam<boundary_id_type>("external_boundary_id");
+  if (isParamValid("external_boundary_name"))
+    params.set<BoundaryName>("external_boundary_name") =
+        getParam<BoundaryName>("external_boundary_name");
+  params.set<MooseEnum>("tri_element_type") = getParam<MooseEnum>("tri_element_type");
+  addMeshSubgenerator("FlexiblePatternGenerator", _input_names[0] + "_periphery", params);
+  _build_mesh = &getMeshByName(_input_names[0] + "_periphery");
```

Comment:
```suggestion
  addMeshSubgenerator("FlexiblePatternGenerator", name() + "_fpgsub", params);
  _build_mesh = &getMeshByName(name() + "_fpgsub");
```
This seems to solve the issue. I do not exactly know why. It may have something to do with the dependency system...

Reply:
That is bizarre. Let me try...

Reply:
I can only think that we have a map somewhere using this name as a key, thus changing the name here affecting the ordering in the map.